annotation-processor/src/main/java/com/linecorp/armeria/server/annotation/processor/DocumentationProcessor.java

```diff
@@ -0,0 +1,186 @@
+
+import com.linecorp.armeria.server.annotation.Description;
+
+/**
+ * Processor that creates a properties file based on the JavaDoc/KDoc description of parameters for interfaces.
+ * This file can be later used if the {@link Description} annotation does not exist for parameters or methods.
+ */
+@SupportedAnnotationTypes({
+        "com.linecorp.armeria.server.annotation.Post",
+        "com.linecorp.armeria.server.annotation.Get",
+        "com.linecorp.armeria.server.annotation.Put",
+        "com.linecorp.armeria.server.annotation.Delete",
+        "com.linecorp.armeria.server.annotation.Head",
+        "com.linecorp.armeria.server.annotation.Options",
+        "com.linecorp.armeria.server.annotation.Patch",
+})
+public class DocumentationProcessor extends AbstractProcessor {
+    private final Map<String, Properties> propertiesMap = new HashMap<>();
+
+    @Override
+    public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+        if (annotations == null || annotations.isEmpty()) {
+            return false;
+        }
+        annotations.forEach(annotation -> processAnnotation(annotation, roundEnv));
+        propertiesMap.forEach((className, properties) -> {
+            try {
+                writeProperties(className, properties);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        });
+        return false;
+    }
+
+    private Properties readProperties(String className) throws IOException {
+        if (propertiesMap.containsKey(className)) {
+            return propertiesMap.get(className);
+        }
+        final FileObject resource = processingEnv
+                .getFiler()
+                .getResource(StandardLocation.CLASS_OUTPUT,
+                             "",
+                             getFileName(className));
+        final Properties properties = new Properties();
+        if (resource.getLastModified() == 0L) {
+            // returns 0 if file does not exist
+            propertiesMap.put(className, properties);
+            return properties;
+        }
+        try (Reader reader = resource.openReader(false)) {
+            properties.load(reader);
+            return properties;
+        }
+    }
+
+    private void writeProperties(String className, Properties properties) throws IOException {
+        if (properties.size() == 0) {
+            return;
+        }
+        final FileObject resource = processingEnv.getFiler().createResource(
+                StandardLocation.CLASS_OUTPUT,
+                "",
+                getFileName(className));
+        try (Writer writer = resource.openWriter()) {
+            properties.store(writer, "Generated list of parameter description of REST interfaces.");
+        }
+    }
+
+    private void processAnnotation(TypeElement annotationElement, RoundEnvironment roundEnv) {
+        roundEnv.getElementsAnnotatedWith(annotationElement)
+                .parallelStream()
```

Comment:
I think we should either just stick to normal `stream` or have a flag to control whether this is parallel or not. Many builds have a lot of parallelism already managed by Gradle so having annotation processors also starting threads can be undesirable.

Reply:
+1 for `stream()`